infer/modules/train/extract/extract_f0_print.py

```diff
@@ -16,19 +16,84 @@
 logging.getLogger("numba").setLevel(logging.WARNING)
 from multiprocessing import Process
 
-exp_dir = sys.argv[1]
-f = open("%s/extract_f0_feature.log" % exp_dir, "a+")
+# exp_dir = sys.argv[1]
+# f = open("%s/extract_f0_feature.log" % exp_dir, "a+")
 
 
 def printt(strr):
     print(strr)
-    f.write("%s\n" % strr)

```

Comment:
Well, print to stdout is also not some good practice. Maybe you can use the logging lib as the other codes did and pass a parameter to make it possible to disable the printing.
We also want to display the log at the web ui, so a custom callback may be needed.

Reply:
Replaced with:
```py
logger = logging.getLogger("f0-feature-extract")

from multiprocessing import Process

# exp_dir = sys.argv[1]
# f = open("%s/extract_f0_feature.log" % exp_dir, "a+")


def printt(strr):
    logger.info(strr)
```
Looks good?

Reply:
The name can be simplified. And you can search about a function hook in logger.